doc/design.md

```diff
@@ -442,3 +500,356 @@ Covers:
+* [`req~auto-create-changelog~1`](system_requirements.md#automatically-create-change-log-entry)
+* [`req~auto-create-pr~1`](system_requirements.md#automatically-create-a-pull-request)
+
+Needs: dsn
+
+##### `dependencies_update.yml` Workflow Receives Vulnerability Info
+`dsn~dependencies_update-vulnerability-info~1`
+
+PK generates the `dependencies_update.yml` workflow so that it receives information about vulnerabilities and issues as optional parameter.
+
+-Needs: impl, utest, itest
+
+Covers:
+* [`dsn~dependencies_update-workflow~1`](#generate-dependencies_updateyml-workflow)
+
+##### `dependencies_update.yml` Workflow Starts PK `update-dependencies` Mode
+`dsn~dependencies_update-starts-pk-update~1`
+
+PK generates the `dependencies_update.yml` workflow so that it starts PK's [`update-dependencies` mode](#update-dependencies-mode), passing information about vulnerabilities.
+
+Rationale:
+
+PK needs the vulnerability info for generating the changelog.
+
+-Needs: impl, utest, itest
+
+Covers:
+* [`dsn~dependencies_update-workflow~1`](#generate-dependencies_updateyml-workflow)
+
+##### `dependencies_update.yml` Workflow Creates a Pull Request
+`dsn~dependencies_update-creates-pull-request~1`
+
+PK generates the `dependencies_update.yml` workflow so that it creates a Pull Request in GitHub. This requires the following steps:
+1. Create a new local branch using a random name
+2. Commit local changes using a commit message that contains the issue number
+3. Push the branch
+4. Create a new pull request with `Closes` comments for each issue number
+5. If the Slack notification URL is available as secret: send a Slack notification
+  * If the workflow fails: send a warning containing the workflow run
+  * If the workflow succeeded: send a success message containing the pull request link
+
+Rationale:
+
+We implement this in a workflow and not in PK because
+* Git/GitHub operations should not be done locally to avoid surprises
+* GitHub action automatically have credentials for pushing and creating a pull request
+* The Slack notification URL might not be available as secret to all repositories, so this step must be optional
+* Sending notifications to developers to
+  * investigate a failed update process
+  * review and merge a new pull request
+
+Note: Implementing this in a workflow makes it hard to do integration tests. We accept that there are no integration tests for running the workflow.
+
+-Needs: impl, utest, itest
+
+Covers:
+* [`dsn~dependencies_update-workflow~1`](#generate-dependencies_updateyml-workflow)
+
+#### Generate `release.yml` workflow
+`dsn~release-workflow~1`
+
+PK generates the `release.yml` GitHub workflow for Maven projects. This workflow runs the build including tests, integration tests, releases to Maven Central and on GitHub.
+
+Rationale:
+* The release process is limited to Maven projects. Support for other projects may be added later.
```

Comment:
Is this a breaking change to release-droids old behavior?

Reply:
and does this effect other projets using release-droid

Reply:
Other projects can continue to use rd as long as the GH workflows are not removed.